jdaviz/core/freezable_state.py

```diff
@@ -54,11 +57,92 @@ def _reset_x_limits(self, *event):
 class FreezableBqplotImageViewerState(BqplotImageViewerState, FreezableState):
     linked_by_wcs = False
 
+    zoom_radius = CallbackProperty(1.0, docstring="Zoom radius")
+    zoom_center_x = CallbackProperty(0.0, docstring='x-coordinate of center of zoom box')
+    zoom_center_y = CallbackProperty(0.0, docstring='y-coordinate of center of zoom box')
+
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         self.wcs_only_layers = []  # For Imviz rotation use.
+        self._during_zoom_sync = False
+        self.add_callback('zoom_radius', self._set_zoom_radius_center)
+        self.add_callback('zoom_center_x', self._set_zoom_radius_center)
+        self.add_callback('zoom_center_y', self._set_zoom_radius_center)
+        for attr in ('x_min', 'x_max', 'y_min', 'y_max'):
+            self.add_callback(attr, self._set_axes_lim)
+        super().__init__(*args, **kwargs)
+
+    @contextmanager
+    def during_zoom_sync(self):
+        self._during_zoom_sync = True
+        try:
+            yield
+        except Exception:
+            self._during_zoom_sync = False
+            raise
+        self._during_zoom_sync = False
+
+    def _set_zoom_radius_center(self, *args):
+        if self._during_zoom_sync or not hasattr(self, '_viewer') or self._viewer.shape is None:
+            return
+        if self.zoom_radius <= 0.0:
+            raise ValueError("zoom_radius must be positive")
+
+        # When WCS-linked (displayed on the sky): zoom_center_x/y and zoom_radius are in sky units,
+        # x/y_min/max are in pixels of the WCS-only layer
+        if self.linked_by_wcs:
+            image, i_ref = get_reference_image_data(self._viewer.jdaviz_app, self._viewer.reference)
+            ref_wcs = image.coords
+            center_x, center_y = ref_wcs.world_to_pixel_values(self.zoom_center_x, self.zoom_center_y)  # noqa
+            center_xr, center_yr = ref_wcs.world_to_pixel_values(self.zoom_center_x+self.zoom_radius, self.zoom_center_y)  # noqa
```

Comment:
I think it is safer to stick to high-level API from https://github.com/astropy/astropy-APEs/blob/main/APE14.rst#high-level-api instead of using the low-level API. Also, I think both `world_to_pixel` and `world_to_pixel_values` can take multiple inputs at once, so it is more performant to vectorize both `center` and `center + radius` inputs in just one call.

Reply:
Is it worth it to pass units around just to get rid of them though?  As long as the low-level API is considered stable, is there harm in using it?

As for combining - how would you suggest doing that so that this is still readable?

Reply:
> Is it worth it to pass units around just to get rid of them though?

Ah, units. Okay, low-level is fine then.

> readable?

I think it would be something like this. Basically you do the look-up once and then unpack later. Look-up can be expensive but unpacking shouldn't be.

```python
coo = ref_wcs.world_to_pixel_values([a bunch of x], [a bunch of y])
center_x  = coo[1]
center_y = ...
```

Reply:
That's what I feared... its just so hard to parse when reading, but if you think its worth it from performance, I can do it

Reply:
For 2 transformations, you could argue that performance does not matter but I cannot tell for sure how much this will be compounded. You can try to benchmark this call with the distorted case from `ImvizDitheredExample` and see.

Reply:
ok, I think I found a way to keep it somewhat readable, let me know what you think

Reply:
LGTM thanks!